crawl-ref/source/main.cc

```diff
@@ -1500,8 +1495,17 @@ static void _take_stairs(bool down)
 
     const bool shaft = (down && get_trap_type(you.pos()) == TRAP_SHAFT);
 
-    if (!(_can_take_stairs(ygrd, down, shaft)
-          && !cancel_barbed_move()
+    if (!_can_take_stairs(ygrd, down, shaft))
+        return;
+
+    if (you.attribute[ATTR_HELD])
+    {
+        free_self_from_net();
+        you.turn_is_over = true;
+        return;
+    }
```

Comment:
Thanks for the welcome!  I have had a patch or two included, but the most recent was in 2012.  I'm amazed at how the build process has evolved.  I work in a small shop (3 programmers); we haven't even implemented automated unit testing.

What you're describing is actually what I did first, but with that you can break the net with < or > even if you aren't on stairs.  I think the code really does need to know if there's nothing else preventing you from using the stairs.  I figured it wouldn't matter if you were barbed or not since you weren't actually going to move when you broke free.

It's debatable about whether the break attempt should happen when you try to enter a shop, use a normal portal, use a transporter or whatever.  I'd chosen to make it work in all those cases because I think it's a better UI experience and I don't see a downside, but maybe not.

It's extra debatable for a transporter.  What do you think of as happening: Is there a button or is it by thought?  ;-)